cmd/upgrade/roles/cmd.go

```diff
@@ -668,12 +663,21 @@ func buildAccountRoleCommandsFromCluster(
 			}
 			_, err = awsClient.IsPolicyExists(policyARN)
 			hasPolicy := err == nil
+			policyAttached := false
+			if hasPolicy {
+				for _, policy := range rolePolicyDetails[accRoleName] {
+					if policy.PolicyArn == policyARN {
```

Comment:
Would you be able to check the second conditional branch in handleAccountRolePolicyARN method ?
It seems it returns the policy arn if the role only has one policy.
This policy can be an arbitrary policy; therefore, it can assume that that is the required policy for the role?

Can you kindly confirm :) 
![image](https://github.com/openshift/rosa/assets/118839428/af5dd0dd-bd52-4f02-8e8a-6355fd978c25)

Reply:
@davidleerh I already removed that condition in this function, also in handleOperatorRolePolicyARN function. That's  one problem this pr fixed as I mentioned in the pr description

```
always find the account/operator-role's default policy by awsClient's function: GetAccountRoleDefaultPolicy/GetOperatorRoleDefaultPolicy. Current logic will regard the only attached policy as the default policy and upgrade it.
```

Reply:
i missed it.
Thank you!